File: Colour the Flag/Colour_the_flag.cpp
#include <bits/stdc++.h>
#define ll long long
using namespace std;
int Colour_the_Flag()
{ int row,col;
  cin>>row>>col;
  vector<string> grid;
  int count_dot = 0;

  for(int r = 0;r < row;r++)
  {  string c;
     cin>>c;
     count_dot += count(c.begin(), c.end(), '.');
     grid.push_back(c);
  }

  if(count_dot == row*col)
  grid[0][0] = 'R';

  for(int i = 0;i < 2;i++)
  {
      for(int r = 0;r < row;r++)
      {
          for(int c = 0;c < col;c++)
          { if(grid[r][c]  != '.')
            {   char next;
                int j;
                // Left 
                if(grid[r][c] == 'R')
                    next = 'W';
                else
                    next = 'R';
                j = c-1;
                while (j >= 0)
                {
                    if(grid[r][j] != '.')
                    {
                        if(grid[r][j] != next)
                        {    cout<<"NO"<<"\n";
                            return (0) ;}
                        else
                            break;
                    }
                    else
                    {grid[r][j] = next;
                     if(grid[r][j] == 'R')
                          next = 'W';
                     else
                         next = 'R';
                        
                    }
                    j -= 1;
                }
                
                
                // Right
                if(grid[r][c] == 'R')
                    next = 'W';
                else
                    next = 'R';
                j = c+1;
                while (j < col)
                {
                    if(grid[r][j] != '.')
                    {
                        if(grid[r][j] != next)
                         {   cout<<"NO"<<"\n";
                            return (0); }
                        else
                            break;
                    }
                    else
                    {grid[r][j] = next;
                     if(next == 'R')
                         next = 'W';
                     else
                         next = 'R';
                        
                    }
                    j += 1;
                }
                
                // top
                if(grid[r][c] == 'R')
                    next = 'W';
                else
                    next = 'R';
                j = r-1;
                while (j >= 0)
                {
                    if(grid[j][c] != '.')
                    {
                        if(grid[j][c] != next)
                          {  cout<<"NO"<<"\n";
                            return (0); }
                        else
                            break;
                    }
                    else
                    {grid[j][c] = next;
                     if(next == 'R')
                         next = 'W';
                     else
                         next = 'R';
                        
                    }
                    j -= 1;
                }
                
                // bottom
                
                if(grid[r][c] == 'R')
                    next = 'W';
                else
                    next = 'R';
                j = r+1;
                while (j < row)
                {
                    if(grid[j][c] != '.')
                    {
                        if(grid[j][c] != next)
                         {   cout<<"NO"<<"\n";
                            return(0) ;}
                         else
                            break;
                    }
                    else
                    {grid[j][c] = next;
                     if(next == 'R')
                         next = 'W';
                     else
                         next = 'R';
                        
                    }
                    j += 1;
                }
                
                
                
                
            }
              
          }
      }
      
      
      
      
      
      
  }
  cout<<"YES"<<"\n";
  for(int r = 0 ;r< row;r++)
  {
      for(int c = 0 ; c < col;c++)
       cout<<grid[r][c];
       cout<<"\n";
  }
  
  return 0;
  
  
  
  
}


int main() {
	int remained_test_cases = 1 ;
	cin>>remained_test_cases;
	while(remained_test_cases > 0)
	{   Colour_the_Flag();
	    remained_test_cases--;
	}
	return 0;
}

File: Colour the Flag/Colour_the_flag.py
def Colour_the_Flag():
    N,M = map(int,input().split())
    grid = []
    count_dot = 0
    # for k in range(2):
    for i in range(N):
        row1 = input()
        count_dot += row1.count(".")
        grid.append(list(row1))
    if count_dot == N*M:
        grid[0][0] = "W"
        
    for k in range(2):
        for i in range(N):
            for j in range(M):
                if grid[i][j] != ".":
                    if grid[i][j] == "W":
                        next = "R"
                    else:
                        next = "W"

                # for left        
                    z = j-1
                    while z >= 0:
                        if grid[i][z] != ".":
                            if grid[i][z] == next:
                                break
                            else:
                               
                                return ["NO",[]]
                        else:
                            grid[i][z] = next
                            if next == "W":
                                next = "R"
                            else:
                                next = "W"
                        z -= 1
              # for right
                    if grid[i][j] == "W":
                            next = "R"
                    else:
                            next = "W"    
                    z = j + 1 

              
                    while z < M:
                        if grid[i][z] != ".":
                            if grid[i][z] == next:
                                break
                            else:
                                
                                return ["NO",[]]
                        else:
                            grid[i][z] = next
                            if next == "W":
                                next = "R"
                            else:
                                next = "W"
                        z += 1
             # for top 
                    if grid[i][j] == "W":
                            next = "R"
                    else:
                            next = "W"
                    z = i-1 
                    while z >= 0:
                        if grid[z][j] != ".":
                            if grid[z][j] == next:
                                break
                            else:
                                
                                return ["NO",[]]
                        else:
                            grid[z][j] = next
                            if next == "W":
                                next = "R"
                            else:
                                next = "W"
                        z -= 1

              # for bottom  
                    if grid[i][j] == "W":
                            next = "R"
                    else:
                            next = "W"
                    z = i+1 
                    while z < N:
                        if grid[z][j] != ".":
                            if grid[z][j] == next:
                                break
                            else:
                                
                                return ["NO",[]]
                        else:
                            grid[z][j] = next
                            if next == "W":
                                next = "R"
                            else:
                                next = "W"
                        z += 1
                    
    return ["YES",grid]        
                
                
remained_test_cases = 1 
remained_test_cases = int(input())
while remained_test_cases  > 0:
    ans = (Colour_the_Flag())
    if ans[0] == "NO":
        print("NO")
    else:
        print("YES")
        for i in range(len(ans[1])):
            row = ""
            for j in range(len(ans[1][0])):
                row += ans[1][i][j]
            print(row)    
            
    
    remained_test_cases -= 1 
    